pkg/runs/runsCancel.go

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package runs
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+
+	"github.com/galasa-dev/cli/pkg/embedded"
+	galasaErrors "github.com/galasa-dev/cli/pkg/errors"
+	galasaapi "github.com/galasa-dev/cli/pkg/galasaapi"
+	"github.com/galasa-dev/cli/pkg/utils"
+)
+
+var (
+	CANCEL_STATUS = "finished"
+	CANCEL_RESULT = "cancelled"
+)
+
+func CancelRun(
+	runName string,
+	timeService utils.TimeService,
+	console utils.Console,
+	apiServerUrl string,
+	apiClient *galasaapi.APIClient,
+) error {
+	var err error
+	var runId string
+
+	log.Printf("CancelRun entered.")
+
+	if runName == "" {
+		err = galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_MISSING_NAME_FLAG, runName)
+	}
+
+	if (err == nil) && (runName != "") {
+		err = ValidateRunName(runName)
+	}
+
+	if err == nil {
+
+		runId, err = getRunIdFromRunName(runName, timeService, apiClient)
+
+		if err == nil {
+
+			updateRunStatusRequest := createUpdateRunStatusRequest(CANCEL_STATUS, CANCEL_RESULT)
+
+			err = cancelRun(runName, runId, updateRunStatusRequest, apiClient)
+
+			if err == nil {
+				consoleErr := console.WriteString(fmt.Sprintf(galasaErrors.GALASA_INFO_RUNS_CANCEL_SUCCESS.Template, runName))
+
+				// Console error is not as important to report as the original error if there was one.
+				if consoleErr != nil && err == nil {
+					err = consoleErr
+				}
+			}
+
+		}
+
+	}
+
+	log.Printf("CancelRun exiting. err is %v", err)
+	return err
+}
+
+func cancelRun(runName string,
+	runId string,
+	runStatusUpdateRequest *galasaapi.UpdateRunStatusRequest,
+	apiClient *galasaapi.APIClient,
+) error {
+	var err error = nil
+	var resp *http.Response
+	var context context.Context = nil
+	var restApiVersion string
+	var responseBody []byte
+
+	restApiVersion, err = embedded.GetGalasactlRestApiVersion()
+
+	if err == nil {
+
+		_, resp, err = apiClient.ResultArchiveStoreAPIApi.PutRasRunStatusById(context, runId).
+			UpdateRunStatusRequest(*runStatusUpdateRequest).
+			ClientApiVersion(restApiVersion).Execute()
+
+		if (resp != nil) && (resp.StatusCode != http.StatusAccepted) {
+			defer resp.Body.Close()
+
+			responseBody, err = io.ReadAll(resp.Body)
+			log.Printf("putRasRunStatusById Failed - HTTP Response - Status Code: '%v' Payload: '%v'", resp.StatusCode, string(responseBody))
```

Comment:
Think you probably want to put a `\n` at the end of the printf string ?

Reply:
Added new line character in case.